src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -52,11 +52,14 @@ public static function castInterval(\DateInterval $interval, array $a, Stub $stu
 
     private static function formatInterval(\DateInterval $i)
     {
-        $format = '%R '
-            .($i->y ? '%yy ' : '')
-            .($i->m ? '%mm ' : '')
-            .($i->d ? '%dd ' : '')
-        ;
+        $format = '%R ';
+
+        if ($i->m >= 12 || $i->d >= 28 || $i->h >= 24 || $i->i >= 60 || $i->s >= 60) {
+            $i = date_diff($d = new \DateTime(), date_add(clone $d, $i)); // recalculate carry over points
```

Comment:
new \DateTime means we change the meaning of the interval: "1 months from now" means something different every month... unless we can really make the representation independant from "now", I think we shouldn't deal with carries.